w3sper.js/src/transaction.js

```diff
@@ -10,53 +10,149 @@ export const TRANSFER =
 import { AddressSyncer } from "./network/syncer/address.js";
 import { Gas } from "./network/gas.js";
 import * as ProtocolDriver from "./protocol-driver/mod.js";
-import { ProfileGenerator } from "./profile.js";
+import { ProfileGenerator, Profile } from "./profile.js";
 
-export class TransactionBuilder {
-  #bookkeeper;
+const _attributes = Symbol("builder::attributes");
 
-  #from;
-  #to;
-  #amount;
-  #obfuscated = false;
-  #gas;
+class BasicTransfer {
+  [_attributes];
 
-  constructor(bookkeeper) {
-    this.#bookkeeper = bookkeeper;
+  constructor(from) {
+    this[_attributes] = Object.create(null);
 
-    this.#gas = new Gas();
+    const value = from instanceof Profile ? { profile: from } : from;
+
+    Object.defineProperty(this, "bookentry", {
```

Comment:
Why defining the property at runtime instead of just assigning it a value?

Reply:
yeah, I was thinking about a getter here, but it's a minor thing anyway: we'll think properly about these cases when we add types.

Reply:
Just to be clear, the scenario I'm referring is when you need to set a property at initialization time that will never be changed from either parties. Although I can use a `getter` here, I usually don't if the private part of the getter will *never* change, even from its own class. Here we're basically set a "constant" that it's valid for the entire lifecycle of the instance, and will never be modified from anyone.

Using the getter you only achieved the part where is not modified by the consumer, but not necessary the provider.